flocker/control/test/test_protocol.py

```diff
@@ -574,6 +575,55 @@ def test_nodestate_notifies_all_connected(self):
             ),
         )
 
+    def test_nodestate_coalesces_multiple_quick(self):
+        """
+        Multiple ``NodeStateCommands`` are coalesced into a single state update
+        broadcast to all the nodes.
+        """
+        self.control_amp_service.configuration_service.save(TEST_DEPLOYMENT)
+
+        agents = [FakeAgent(), FakeAgent()]
+        clients = list(AgentAMP(Clock(), agent) for agent in agents)
+        servers = list(LoopbackAMPClient(client.locator) for client in clients)
+
+        for server in servers:
+            self.control_amp_service.connected(server)
+
+        initial_update_counts = list(
+            agent.cluster_updated_count for agent in agents)
+
+        for i in xrange(10):
+            new_state = NODE_STATE.set(
+                'applications',
+                NODE_STATE.applications.add(
+                    Application(name=('app-%d' % i),
+                                image=DockerImage.from_string('image-%d' % i))
+                )
+            )
```

Comment:
:question: So this is like a single client sending 10 state updates instantly. 
 * Is that realistic? 
 * Does it matter if it's not realistic?
 * Wouldn't this be coalesced by the only-send-update-if-agent-acknowledged-the-last-update optimization, rather than the delay-update-by-1s optimization?